lms/views/api/canvas/authorize.py

```diff
@@ -41,11 +41,17 @@
 )
 
 
+# Support for creating assignments using Canvas Pages
 PAGES_SCOPES = (
     "url:GET|/api/v1/courses/:course_id/pages",
     "url:GET|/api/v1/courses/:course_id/pages/:url_or_id",
 )
 
+# All the scopes that our LMS app may use.
+ALL_SCOPES = (
+    FILES_SCOPES + FOLDERS_SCOPES + GROUPS_SCOPES + PAGES_SCOPES + SECTIONS_SCOPES
+)
+
```

Comment:
Maybe we could avoid the potential for this `ALL_SCOPES` list to get out of date when another `FOO_SCOPE` constant is added by changing them all to a dict:

```python
SCOPES = {
    "files": (
        "url:GET|/api/v1/courses/:course_id/files",
        "url:GET|/api/v1/files/:id/public_url",
    ),
    "folders": (...),
    ...
}

ALL_SCOPES = set(scope for scopes in SCOPES.values() for scope in scopes)
```

Incidentally this also removes the possibility of duplicate scopes being listed.